Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Use the official Rust image as the base image
+FROM rust:1.81 AS builder
+
+# Install protobuf compiler
+RUN apt-get update && apt-get install -y protobuf-compiler
+
+# Create a new directory for the application
+WORKDIR /app
+
+# Copy the entire project
+COPY . .
+
+# Build dependencies and the project
+RUN cargo build --release
+
+# Create a new stage with a minimal image
+FROM ubuntu:22.04 AS final
```

Comment:
Maybe here is opinionated but we should have a slim final version so we don't have that large images for deployments.